src/query/graphite/native/builtin_functions.go

```diff
@@ -200,6 +200,23 @@ func limit(_ *common.Context, series singlePathSpec, n int) (ts.SeriesList, erro
 	return r, nil
 }
 
+// grep takes a metric or a wildcard seriesList, followed by a regular
+// expression in double quotes. Excludes metrics that don’t match the regular expression.
+func grep(_ *common.Context, seriesList singlePathSpec, regex string) (ts.SeriesList, error) {
+	re := regexp.MustCompile(regex)
+
+	var output []*ts.Series
```

Comment:
Can actually do this without allocations by filtering inplace like this:
```golang
filtered := seriesList.Values[:0]
for _, series := range seriesList.Values {
	if re.MatchString(series.Name()) {
		filtered = append(filtered, series)
	}
}

r.Values = filtered[:len(filtered)]
```

Reply:
good call!

Reply:
done